trade-finance-manager-ui/server/controllers/case/index.js

```diff
@@ -315,6 +315,14 @@ const getCaseFacility = async (req, res) => {
     deal.tfm.stage = DEAL.DEAL_STAGE.AMENDMENT_IN_PROGRESS;
   }
 
+  /**
+   * Ensure imperative deal properties exist before rendering
+   */
+  if (!deal?.dealSnapshot?._id || !deal?.tfm) {
+    console.error('An error occurred while rendering a TFM deal %s', dealId);
+    return res.render('_partials/problem-with-service.njk');
+  }
```

Comment:
I think this should be directly after line 307 - line 310 and 315 use `deal.tfm`

Reply:
Can we also add a unit test for this please?

Reply:
Yes good spot, I have moved it straight after `const deal = await api.getDeal(dealId, userToken);`.